### ISSUE TITLE
Make it work in CI

### ISSUE THREAD

**Issue by dpc** [crev-dev/cargo-crev#131]
* [x] fix all the issues that prevent it
* [x] `--for-id <id>` argument to `cargo crev verify`; the core logic is already there
* [x] status code on `cargo crev verify deps` 0 - all deps passed, -1 - some stuff didn't pass, -2 error
* [ ] prepare example for other people to use (`cargo crev fetch all && cargo crev verify deps -for-id`?)

**Comment by dpc** [comment]
Needs more testing, but hopefully it's working now.

**Comment by dpc** [comment]
It would be cool if `cargo crev verify` could save its state somewhere, so that verification would fail only the number of unverified deps increased. This way projects could start incrementally.

**Comment by nbigaouette-eai** [comment]
Hi dpc! I'm trying out cargo-crev and would like to add it in our CI.

Let's say I've reviewed myself a couple of crates and published my reviews (see https://github.com/nbigaouette-eai/crev-proofs). Possibly some colleagues could do the same.

If I want my CI to run the verification, I obviously don't want to create a `CrevID` for the CI. From what I understand from this issue, I should be able to run a verification by trusting specific users (including myself)?

Here's what I have for now:
```
# Fetch all the proofs from the author of crev ("dpc")
cargo crev fetch url https://github.com/dpc/crev-proofs
# Fetch our proofs
cargo crev fetch url https://github.com/nbigaouette-eai/crev-proofs
# Verify from nbigaouette-eai
cargo crev verify --for-id CK6KiH-fFFiHCHOmG7bNOHooVL8ujCavQ7yfkcqxUDQ
```

Does that make sense? If we are many doing code reviews, how can we let the CI verify "from multiple ids"?

Thanks!

**Comment by dpc** [comment]
Sounds OK. Though I would just fetch the root id, and then recursively fetch all others with `fetch trusted`.


```
cargo crev fetch url https://github.com/nbigaouette-eai/crev-proofs
cargo crev fetch trusted # recursively fetch all the repos you transitively trust
cargo crev verify --for-id CK6KiH-fFFiHCHOmG7bNOHooVL8ujCavQ7yfkcqxUDQ
```

In a organization settings you might want to create a CrevId, just for the purposes of CI, and make that CrevId trust whoever you feel like you want to trust. 

In case something is not working, let me know, we can chat on matrix, etc. and I'm very happy to help and discuss any improvements that would make sense.

**Comment by dpc** [comment]
@nbigaouette-eai How did it go? :)

**Comment by nbigaouette-eai** [comment]
I unfortunately had to revert back to the steps I posted; crev was panicking :(
```
# Fetch our proofs
cargo crev fetch url https://github.com/nbigaouette-eai/crev-proofs
# Recursively fetch all the repos we transitively trust
cargo crev fetch trusted
# Verify from nbigaouette-eai
cargo crev verify --for-id CK6KiH-fFFiHCHOmG7bNOHooVL8ujCavQ7yfkcqxUDQ

Fetching https://github.com/nbigaouette-eai/crev-proofs... OK; 2 new trust proofs; 1 new package reviews
Found proofs from:
       3 CK6KiH-fFFiHCHOmG7bNOHooVL8ujCavQ7yfkcqxUDQ
Error: Current Id not set
stack backtrace:
   0: failure::backtrace::internal::InternalBacktrace::new::h1fcee5d15e5d659d (0x1040ce0)
   1: failure::backtrace::Backtrace::new::hb98ce47facf0b678 (0x1040ea0)
   2: crev_lib::local::Local::fetch_trusted::h16cd2ec4e60de520 (0xce536f)
   3: cargo_crev::run_command::hc72bfd5541557311 (0x41b414)
   4: cargo_crev::main::h60f83c21e576a7b1 (0x41eb93)
   5: std::rt::lang_start::{{closure}}::hf8bd4a72133e2113 (0x4cb1e3)
   6: std::rt::lang_start_internal::{{closure}}::h52e30c7908fe68b7 (0x10901b3)
             at src/libstd/rt.rs:49
      std::panicking::try::do_call::h2ea8cd7434cffd4d
             at src/libstd/panicking.rs:293
   7: __rust_maybe_catch_panic (0x10997ea)
             at src/libpanic_unwind/lib.rs:85
   8: std::panicking::try::h549742e035b8a472 (0x1090c1b)
             at src/libstd/panicking.rs:272
      std::panic::catch_unwind::h406d801f678c0df0
             at src/libstd/panic.rs:394
      std::rt::lang_start_internal::h4b30ec1a8a6b305f
             at src/libstd/rt.rs:48
   9: main (0x41eca2)
Exited with code 254
```

**Comment by dpc** [comment]
OH. Silly me. `cargo crev fetch trusted` requires a current id, otherwise how would it know which IDs are trusted (by whom?).

You can do `cargo crev fetch all` instead, I believe. And I'll try to add `--for-id` to `fetch trusted` as well.

**Comment by dpc** [comment]
@nbigaouette-eai Implemented in 965f9ea

**Comment by nbigaouette-eai** [comment]
I've updated my Cargo.lock format which broke cargo-crev (0.11) so I've updated it (to 0.16.1) and revisiting this.

What I'd like to do is:
1. Create locally my id (`CK6KiH-fFFiHCHOmG7bNOHooVL8ujCavQ7yfkcqxUDQ`). Done.
2. Trust some high profile rust users, for example you and others. I believe this adds the trust to my own repo, right?
3. Locally perform some review and publish them. That should push these review to my own repo, right?
4. Add a step in my CI that will trust my own id without having to define an id for the CI.
5. Perform a verification where the chain of trust starts from my own id.

I currently have the following in my CI:
```shell
cargo crev repo fetch url https://github.com/dpc/crev-proofs
cargo crev repo fetch url https://github.com/nbigaouette-eai/crev-proofs
cargo crev crate verify --show-all --for-id CK6KiH-fFFiHCHOmG7bNOHooVL8ujCavQ7yfkcqxUDQ
```
but I think it's probably outdated or slightly wrong. I should not have to fect both repo after having done Step 2. above, no?

What would be the best way to perform a verification starting trust at my own id?

Thanks! :D
